docs/plugin-development.rst

```diff
@@ -508,6 +508,41 @@ login via a username and password. This allows alternative authentication
 modes to be used instead of core, or prior to attempting core authentication.
 The event info contains two keys, "login" and "password".
 
+Customizing the Swagger page
+****************************
+
+To customize text on the Swagger page, create a
+`Mako template <http://www.makotemplates.org/>`_ file that inherits from the
+base template and overrides one or more blocks:
+
+.. code-block:: html+mako
+
+    <%inherit file="${context.get('baseTemplateFilename')}"/>
```

Comment:
Why do you need to use `${context.get('baseTemplateFilename')}`?  Since `self.vars` is injected directly into the `template.render` call, can't you just reference it as `${baseTemplateFilename}`?

Reply:
Mako treats directives like `<%inherit>` differently from the body content. http://docs.makotemplates.org/en/latest/runtime.html#context-variables describes that
>  variable names that are referenced within the function [...] are pulled from the Context object’s dictionary
Looking at the compiled template, the function generated for the inherit directive doesn't pull variables out of the context, so it has to be done manually:
```python
def _mako_inherit(template, context):
    _mako_generate_namespaces(context)
    return runtime._inherit_from(context, (context.get('baseTemplateFilename')), _template_uri)
```